hashintel/hash
# Minimal Ory kratos configuration

## 🌟 What is the purpose of this PR?




This PR adds a minimal configuration of [Ory Kratos](https://www.ory.sh/docs/kratos/) to the HASH workspace, replacing the previously implemented signup and login flows.

Please note this configuration is missing some important components of authentication (such as email verification flows, creating a user entity in the HASH graph when a kratos identity is registered, etc). These will be addressed in follow-up.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1202538466812818/1202655882079016/f) _(internal)_

## 🔍 What does this change?




### HASH workspace dev environment
- fixed the `yarn dev` command to work without the (now deleted) hash workspace pg instance
- updated the HASH workspace dev docker-compose script to:
  - create a `kratos` database in the postgres instance
  - run the ory kratos image (in `--dev` mode)
  - run other helpful kratos related images (`kratos-migrate`,`mailslurper`)

### HASH workspace BE
- replaced express-session/passport workspace API middleware with kratos session management

### HASH workspace FE
- replaced the previously implemented signup page with a minimal kratos signup form (email + password)
- replaced the previously implemented login page with a minimal kratos login form (email + password)
- replaced the previously implemented homepage with a page that tells you if you are logged in with kratos (for dev purposes)

## ⚠️ Known issues



## 🐾 Next steps



- [add kratos email verification flow](https://app.asana.com/0/0/1202714661072442/f)
- [add kratos account recovery flow](https://app.asana.com/0/0/1202714892950391/f)
- [create a user entity in the graph when the kratos registration flow is completed](https://app.asana.com/0/1202621375435411/1202693393444197/f)
- [create a kratos user identity when seeding the dev environment for testing/dev purposes
](https://app.asana.com/0/0/1202714661072444/f)
- [Replace bcrypt with argon2 hashing algorithm in kratos config](https://app.asana.com/0/0/1202721575417478/f)

## 🛡 What tests cover this?



None. We should add FE tests for the auth flows when they are in a more finalised state (when we have verification steps, user shortname, etc in place).


## ❓ How to test this?



1. Checkout the branch
2. Install the latest dependencies (`yarn install`)
3. Run the HASH workspace external services (`yarn external-services up`)
4. run the HASH workspace (`yarn dev`)
5. Open http://localhost:3000 in the browser. You should be redirected to the login page, where you can login or redirect to the signup page.

## 📹 Demo




https://user-images.githubusercontent.com/42802102/182638962-4528c609-677f-4a7c-a5bf-31b9b5c70cda.mp4
